src/Generation/EmulatorSupportGenerator.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+declare(strict_types=1);
+
+namespace Google\Generator\Generation;
+
+use Google\Auth\Credentials\InsecureCredentials;
+use Google\Generator\Ast\AST;
+use Google\Generator\Ast\Access;
+use Google\Generator\Ast\PhpDoc;
+use Google\Generator\Utils\Type;
+use \Grpc\ChannelCredentials;
+
+class EmulatorSupportGenerator
+{
+    /**
+     * Map of emulator support required clients and their expected env variables.
+     */
+    private static $emulatorSupportFixes = [
```

Comment:
Don't call this "Fixes". We are not fixing anything here, we are adding a feature (emulator support). Would be better to just call it something like `emulatorSupportClients`, since the keys are client names.

Reply:
Sounds great.